docs/source/discover.rst

```diff
@@ -1,9 +1,54 @@
+.. py:module:: kasa.discover
+
 Discovering devices
 ===================
 
 .. contents:: Contents
    :local:
 
+Discovery
+*********
+
+Discovery works by sending broadcast UDP packets to two known TP-link discovery ports, 9999 and 20002.
+9999 is for legacy devices that didn't use strong encryption and 20002 is for newer devices that use different
+levels of encryption.
+If a device uses 20002 for discovery it will return some basic information with sensitive data hashed.
```

Comment:
```suggestion
If a device uses port 20002 for discovery, you will still obtain some basic information from the device, but you need to pass credentials for complete response.
```
Maybe something like this?

Reply:
Updated but with additional information.